OpenRA.Mods.Common/Traits/Harvester.cs

```diff
@@ -183,8 +183,8 @@ public Actor ClosestProc(Actor self, Actor ignore)
 
 					var occupancy = refs[loc].Occupancy;
 
-					// 4 harvesters clogs up the refinery's delivery location:
-					if (occupancy >= 3)
+					// 3 harvesters clogs up the refinery's delivery location:
+					if (occupancy >= 2)
```

Comment:
When you have tested it you will note that Harvesters will wait on the Field until the refinery is free for the next Harvester. the changes i have made minutes ago will filter out blocked refineries.